src/main/kotlin/app/manager/service/ManagerService.kt

```diff
@@ -91,39 +91,48 @@ class ManagerService
 
             val bucketResponseEntity = assetStoreApi.getSnippet(snippetKey)
 
+            // Missing getting the author from the permissions service
+
             if (bucketResponseEntity.statusCode.is2xxSuccessful) {
                 val content = bucketResponseEntity.body!!
                 return GetSnippetOutput(
+                    id = snippetId,
                     name = snippet.name,
                     content = content,
+                    language = snippet.language,
+                    author = "authorId",
                 )
             } else {
                 throw NotFoundException("Failed to get snippet. Status code: ${bucketResponseEntity.statusCode}")
             }
         }
 
-        fun getSnippetsFromUserId(userId: String): List<GetAllSnippetsOutput> {
+        fun getSnippetsFromUserId(userId: String): List<GetSnippetOutput> {
             val permissionResponseEntity =
                 snippetPermissionApi.getAllSnippetsPermission(userId)
 
             if (permissionResponseEntity.statusCode.is2xxSuccessful) {
-                val snippets: MutableList<GetAllSnippetsOutput> = emptyList<GetAllSnippetsOutput>().toMutableList()
+                val snippets: MutableList<GetSnippetOutput> = emptyList<GetSnippetOutput>().toMutableList()
                 for (permissionSnippet in permissionResponseEntity.body!!) {
                     val snippetId = permissionSnippet.snippetId
                     val snippetAuthor = permissionSnippet.authorId
                     val snippet = snippetRepository.findSnippetById(snippetId) ?: throw Exception("Snippet not found")
+                    val content =
+                        assetStoreApi.getSnippet(snippet.snippetKey).body
+                            ?: throw Exception("Failed to get snippet content")
```

Comment:
We should start throwing better errors to help debugging. Maybe add the error response from the assetService?